### PR TITLE
Expose metrics

### PR DESCRIPTION
**Pull request by chhetripradeep**
Closes https://github.com/pavolloffay/jaeger-clickhouse/issues/19

Adds metrics for batch size and flush interval along with their count in prometheus exposition format.

Signed-off-by: Pradeep Chhetri <pradeepchhetri4444@gmail.com>

### CODE REVIEW

File: internal/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package metrics
```
Comment (pavolloffay) [inline internal/metrics/metrics.go:1]: I would prefer to keep this file in the same package as writer.go.

File: internal/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+// https://prometheus.io/docs/practices/naming/
+var (
+	ClickHouseBatchSize = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "jaeger_clickhouse_batch_size",
+		Help: "Maximum size of a batch written to clickhouse",
+	})
+	ClickHouseFlushInterval = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "jaeger_clickhouse_flush_interval_seconds",
```
Comment (pavolloffay) [inline internal/metrics/metrics.go:16]: is the seconds granularity enough?

File: internal/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// https://prometheus.io/docs/practices/naming/
+var (
+	ClickHouseBatchSize = prometheus.NewGauge(prometheus.GaugeOpts{
```
Comment (pavolloffay) [inline internal/metrics/metrics.go:11]: I am not sure if this is a useful metric. Who runs Jaeger ultimately knows what is the setting. 

The same applies to ClickHouseFlushInterval.

File: internal/metrics/metrics.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	})
+	NumWritesWithFlushInterval = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "jaeger_clickhouse_writes_with_flush_interval_total",
+		Help: "Number of clickhouse writes due to flush interval criteria",
+	})
+)
+
+var registerMetrics sync.Once
+
+func Register() {
```
Comment (pavolloffay) [inline internal/metrics/metrics.go:31]: Can we avoid the global state and pass p8s instance/metric registry to `NewWriter` and register it there?

File: storage/config.go [hunk 5]
```diff
@@ -20,4 +20,6 @@ type Configuration struct {
 	Address string `yaml:"address"`
 	// Directory with .sql files that are run at plugin startup.
 	InitSQLScriptsDir string `yaml:"init_sql_scripts_dir"`
+	// Url for exposing prometheus metrics e.g. http://localhost:1000.
+	MetricsURL string `yaml:"metrics_url"`
```
Comment (pavolloffay) [inline storage/config.go:24]: Could we rename it to `MetricsEndpoint`? 

endpoint is more used in OTELcol and Jaeger will be rebased on top of it soon.

File: config.yaml [hunk 6]
```diff
@@ -1,9 +1,12 @@
...
 init_sql_scripts_dir:
-# Batch size. Default 10_000.
+# Batch write size. Default 10_000.
 batch_write_size:
-# Batch flush size. Default 5s.
+# Batch flush interval. Default 5s.
 batch_flush_interval:
 # Encoding of stored data. Either json or protobuf. Default json.
 encoding:
+# Url for exposing prometheus metrics.
```
Comment (pavolloffay) [inline config.yaml:11]: Is there any default value?

**Comment by pavolloffay** [comment]
@chhetripradeep could you please rebase?

**Comment by chhetripradeep** [comment]
Oops. Sorry my mistake. Doing it now.

**Comment by chhetripradeep** [comment]
@pavolloffay Thank you for the review. Sorry for delay on my side. It is ready for review again.

**Comment by chhetripradeep** [comment]
I think with rebase something break. Fixing it.

**Comment by chhetripradeep** [comment]
Also rolled back the docker networking changes merged in earlier PR. It works as expected now. It is ready for review again.

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**